### DISCUSSION TITLE
Welcome to esp32-cam-webserver Discussions!

### DISCUSSION THREAD

**Discussion by easytarget** [easytarget/esp32-cam-webserver#161] [Announcements]
## 👋 Welcome!
  I'm using Discussions as a place to connect with other people for the questions/discussions that do not really belon in th eissues database,
  * Ask questions you’re wondering about.
  * Share what you have done with your camera, interesting mountings, etc!
  * Share ideas.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what is on your mind.

**Comment by Enolyp** [comment]
Hello,

First, thanks for your work, it's really helpful :)

But i have a question about modify the User Interface.
You did a great job at reducing the UI. I want to modify the UI too (like the change the color of the button for example), but i don't know how to do it.

Can you enlight me, sir ?

Thanks a lot.

-Eno-

**Comment by easytarget** [comment]
> Can you enlight me, sir ?

To be honest, this probably isnt the place for that.

The longer reply is that modifying the UI needs knowlege of (at minimum) CSS and HTML. If you have those skills the CSS is in `css.h` and the the various html pages are defined in `ov_*.h`. and should be straightforward.
* If you don't have basic HTML and CSS knowlege my recommended starting point would be https://www.w3schools.com/ .
* The UI also uses JavaScript quite extensively, and advanced modifications would need knowledge of that.

However: Doing *very* simple color and/or font modifications isn't too hard: 
[css-blue.h.txt](https://github.com/easytarget/esp32-cam-webserver/files/7322490/css-blue.h.txt)

If you rename this to 'css.h' in the main sketch folder, recompile and force reload (press 'ctrl while hitting the browser reload button' to avoid using the cached css file in the browser) you will get something like:
![Screenshot from 2021-10-11 14-20-20](https://user-images.githubusercontent.com/4005142/136794577-b433c6ef-4252-4846-9bc6-ed8bff71d5fa.png)

All that I have done here is edited the color for the buttons to from red to blue (the background color!) and the main page background from slate-grey to off-white, while doing the opposite to the main page font color. If you compare the two files you can see that this had to be done in several places, and knowing/understanding how this works is beyond the scope I want to cover here. But would make a good place for a beginner to start..

For reference here is a list of differences (the numbers above each block are the line numbers, the `<` and `>` show you which file which line (original is left, blue version is right). Colors are being specified as RGB hexadecimal pairs.
```
[owen@hobbes esp32-cam-webserver]$ diff css.h css-blue.h 
11,12c11,12
<   background: #181818;
<   color: #EFEFEF;
---
>   background: #EFEFEF;
>   color: #181818;
125c125
<   background: #ff3034;
---
>   background: #3030ff;
132c132
<   background: #ff494d
---
>   background: #3080ff
136c136
<   background: #f21c21
---
>   background: #30e0ff
171c171
<   background: #ff3034;
---
>   background: #3030ff;
195c195
<   background: #ff3034;
---
>   background: #3030ff;
225c225
<   background: #ff3034;
---
>   background: #3030ff;
287c287
<   background-color: #ff3034
---
>   background-color: #3030ff
312c312
<   background: #ff3034;
---
>   background: #3030ff;
```

**Reply by Enolyp** [reply]
Thanks a lot. 

During the time of your answer, i did research and find how to modify the UI, and i succeed to do what i want.

Have a nice day :)

**Comment by nwaehner** [comment]
Hi! I'm Nicolas, and I'm a physics student doing research in a lab in Buenos Aires.

I'm trying to obtain maximum resolution images from an OV3660 (which is 2048x1564), but when i download the image from the webserver it crops the image with lower resolution. 

Do you know if it's possible to save images with full resolution? I haven't found any code out there. That's all i want, and if it can be in .raw format it would be ideal. 

Thank u for your time.
Nicolas.
